File: leetcode/js/112-path-sum.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} targetSum
 * @return {boolean}
 */
var hasPathSum = function(root, targetSum) {
    if (root === null) {return false;}

    let sum = targetSum - root.val;

    if ((sum === 0) && (root.left === null) && (root.right === null)) {
        return true;
    }
    else {     
        return (hasPathSum(root.left, sum) || hasPathSum(root.right, sum));
    }
};

File: leetcode/py/112-path-sum.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def hasPathSum(self, root, targetSum):
        if not root: return False
        t = targetSum - root.val
        
        if t == 0 and root.left == None and root.right == None:
            return True
        else:
            return self.hasPathSum(root.left, t) or self.hasPathSum(root.right, t)